aerogear/mobile-security-service-operator
# Doing fixes and refacs in the code

## Motivation
- A comment raised by a colleague across the teams. 
- https://github.com/aerogear/mobile-security-service-operator/pull/81
- https://issues.jboss.org/browse/AEROGEAR-9231
- https://issues.jboss.org/browse/AEROGEAR-9213
- https://issues.jboss.org/browse/AEROGEAR-9156

## What

- Fix impl to update the status of the Services in the CR. (introduced at https://github.com/aerogear/mobile-security-service-operator/pull/81)
- Fix impl to create the values of the SDKConfigMap  (introduced at https://github.com/aerogear/mobile-security-service-operator/pull/81)
- Keep the reconcile calls centralized in the reconcile func in order to make the code more understandable, clean and follow the same standard proposed in the operator-framework docs and examples.
- NOT do the re-queue when an object is created/updated always as it was since the reconcile calls were not centralized as described in the item above.
- Remove return of error since has no error scenario in this case that could be returned
- Fix code format by using `make fmt`
-  add setup to githooks
- fix export of variables locally for the run-local command.

## Why
- Solve issues
- Good Practices
- Make the code more understandable.

## Verification Steps
* Use the image `docker.io/cmacedo/mobile-security-service-operator:REF-RECONCILE`
* Run `make create-all` and ensure that all will be installed and get running with success.

### Check the  MSS CR Status
* Go to `Resources >> Other Resources` and check that has a proxyServiceName and  proxyServiceStatus for the MSS CR. Check if the names of the ServiceName and of ProxyServiceName are correct.
<img width="896" alt="Screenshot 2019-05-21 at 09 18 38" src="https://user-images.githubusercontent.com/7708031/58079857-e305ac00-7ba9-11e9-9474-540eae15bc2a.png">

### Check the API URL in the SDKConfigMap

* Create the namespace `mobile-security-service-apps` 
* Create the app by using `make create-app`
* Check if the SDKConfigMap of this app has as URL the route + API_ENDPOINT
* Check that the app was created in the Service

<img width="1467" alt="Screenshot 2019-05-21 at 09 19 56" src="https://user-images.githubusercontent.com/7708031/58079866-e8fb8d00-7ba9-11e9-8548-a2b5e6172dd3.png">
<img width="1455" alt="Screenshot 2019-05-21 at 09 19 29" src="https://user-images.githubusercontent.com/7708031/58079885-f4e74f00-7ba9-11e9-8c1e-b831f004223a.png">
<img width="1529" alt="Screenshot 2019-05-21 at 09 19 18" src="https://user-images.githubusercontent.com/7708031/58079893-fa449980-7ba9-11e9-91a3-b70ecf7b39b2.png">
<img width="1467" alt="Screenshot 2019-05-21 at 09 19 56" src="https://user-images.githubusercontent.com/7708031/58079904-ff094d80-7ba9-11e9-83eb-ca238f1a1477.png">

### Check delete an app
* Run `make delete-all`
* Check that the app was removed from the Service

<img width="1122" alt="Screenshot 2019-05-21 at 09 23 04" src="https://user-images.githubusercontent.com/7708031/58079928-0e889680-7baa-11e9-9708-2dbc5d5f02ef.png">

**NOTE:** The delete all worked perfectly. 

<img width="1045" alt="Screenshot 2019-05-21 at 09 23 52" src="https://user-images.githubusercontent.com/7708031/58080010-37a92700-7baa-11e9-8549-45201f0c7366.png">

## Checklist:


## Progress


## Additional Notes
